### PR TITLE
Fix(atlantis) Unable to use custom port #420

### LINKED ISSUE #420: Unable to use custom port
You have the ability to set the `targetPort` and `portName` on the service definition, as well as the ability to define `ATLANTIS_PORT` env variable on the `statefulSet` to use a custom port with this workload. 

However, this does not update the port on container spec or the readinessProbes. 

These sections use static blocks 
https://github.com/runatlantis/helm-charts/blob/main/charts/atlantis/templates/statefulset.yaml#L238-L240
https://github.com/runatlantis/helm-charts/blob/main/charts/atlantis/templates/statefulset.yaml#L505-L521

Propose updating the chart to use `service.targetPort` to configure all of the necessary port settings on the chart resources.

Ill cut an MR.

### PR DESCRIPTION
**Pull request by doctahpopp**
## what

- Provide ability to use the `service.TargetPort` key to define alternative ports for running atlantis.
- Extended documentation for the `targetPort` key to include implications.

## why

- There is partial support for custom ports, but it is not extended to include all port related configurations. This allows for a hollistic approach to custom ports.

## tests

Ive extended the helm tests to cover this use-case.

## references

`closes #420 `

### CODE REVIEW

**Comment by GMartinez-Sisti** [comment]
Hi @doctahpopp, thanks for the contribution. Can you please also bump the minor chart's version (since it's a new feature)? 🙏

**Comment by doctahpopp** [comment]
On it!

File: charts/atlantis/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: v1
 appVersion: v0.29.0
 description: A Helm chart for Atlantis https://www.runatlantis.io
 name: atlantis
-version: 5.5.1
+version: 5.6.1
```
Comment (GMartinez-Sisti) [inline charts/atlantis/Chart.yaml:6]: It's common to reset patch when minor is increased: 🙂
```suggestion
version: 5.6.0
```

Comment (doctahpopp) [reply charts/atlantis/Chart.yaml:6]: That's what I get for rushing

Comment (doctahpopp) [reply charts/atlantis/Chart.yaml]: Alll Set! squashed my commits too.

**Comment by doctahpopp** [comment]
@GMartinez-Sisti is there anything yall need from me here for the failed test ? I see its a UI test failure, but Im not sure what might be causing that.

**Comment by GMartinez-Sisti** [comment]
> @GMartinez-Sisti is there anything yall need from me here for the failed test ? I see its a UI test failure, but Im not sure what might be causing that.

I need to reproduce this locally and troubleshoot to figure out what's happening. Best guess is that it's a matter of timing, but needs confirmation.

**Comment by GMartinez-Sisti** [comment]
This was probably the issue -> https://github.com/runatlantis/helm-charts/pull/425#issuecomment-2369782446 and just merged. Please fix the conflicts and rebase from main so we can check if the CI is able to work now.

**Comment by doctahpopp** [comment]
@GMartinez-Sisti I think we should be good here finally!

**Review by GMartinez-Sisti — APPROVED** [review]

**Pull request merged**